flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/Execution.java

```diff
@@ -1061,6 +1063,20 @@ void markFinished(Map<String, Accumulator<?, ?>> userAccumulators, IOMetrics met
 							IntermediateResultPartition[] allPartitions = finishedPartition
 									.getIntermediateResult().getPartitions();
 
+							PartitionTable<ResourceID> partitionTable = vertex.getExecutionGraph().getPartitionTable();
+							for (IntermediateResultPartition partition : allPartitions) {
+								ResultPartitionDeploymentDescriptor descriptor = producedPartitions.get(partition.getPartitionId());
+								if (!descriptor.isReleasedOnConsumption()) {
+									if (!taskManagerLocationFuture.isDone()) {
```

Comment:
The execution was already finished then it must have a assigned slot before, so I think it is not necessary here. Otherwise we should make this check in the beginning of markFinished, not in the process of starting tracking partitions.